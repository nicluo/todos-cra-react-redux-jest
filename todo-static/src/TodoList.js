import React, { Component } from 'react';
import TodoItem from "./TodoItem";
import TodoForm from "./TodoForm";

const TODOS = [
  {title: "Start With A Mock", completed: true},
  {title: "Step 1: Break The UI Into A Component Hierarchy", completed: true},
  {title: "Step 2: Build A Static Version in React", completed: true},
  {title: "Step 3: Identify The Minimal (but complete) Representation Of UI State", completed: false},
];

const TODOFORM = {
  input: "New Todo Item"
};

class TodoList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: TODOS,
      todoForm: TODOFORM
    };
  }

  render() {
    const {todos, todoForm} = this.state;

    const todoListContent = todos.map(function(todo) {
      return (<TodoItem completed={todo.completed}>{todo.title}</TodoItem>);
    });

    return (
      <div className="TodoList ph3 mv4">
        {todoListContent}
        <TodoForm {...todoForm} />
      </div>
    );
  }
}

export default TodoList;
