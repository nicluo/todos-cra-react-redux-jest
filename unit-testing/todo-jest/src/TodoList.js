import React, { Component } from 'react';
import TodoItem from "./TodoItem";
import TodoForm from "./TodoForm";

const DEFAULT_STATE = {
  todos: [],
  input: ""
};

class TodoList extends Component {
  state = DEFAULT_STATE;

  handleInputChange = (event) => {
    this.setState({input: event.target.value});
  };

  handleSubmit = (event) => {
    this.setState((prevState) => {
      const newTodos = prevState.todos.slice();
      const todo = {
        // TODO: Generate better unique ids rather than using Math.random()
        id: Math.floor(Math.random() * 1000),
        title: prevState.input,
        completed: false
      };
      newTodos.push(todo);
      return {todos: newTodos, input: ''};
    });
    event.preventDefault();
  };

  handleTodoChange = (id, completed) => {
    this.setState((prevState) => {
      const newTodos = this.state.todos.map((todo) => {
        if (todo.id === id) {
          todo.completed = completed
        }
        return todo;
      });
      return {todos: newTodos};
    });
  };

  render() {
    const {todos, input} = this.state;

    const todoListContent = todos.map((todo) => {
      return (<TodoItem key={todo.id} id={todo.id} completed={todo.completed} onChange={this.handleTodoChange}>{todo.title}</TodoItem>);
    });

    return (
      <div className="TodoList ph3 mv4">
        {todoListContent}
        <TodoForm onChange={this.handleInputChange} onSubmit={this.handleSubmit} input={input} />
      </div>
    );
  }
}

export default TodoList;
