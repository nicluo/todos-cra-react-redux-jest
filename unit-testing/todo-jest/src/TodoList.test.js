import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from './TodoList';

it('Renders initial state without crashing', () => {
  const component = renderer.create(
    <TodoList />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});