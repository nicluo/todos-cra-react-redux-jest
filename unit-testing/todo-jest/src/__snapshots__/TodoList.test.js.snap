// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders initial state without crashing 1`] = `
<div
  className="TodoList ph3 mv4"
>
  <div
    className="TodoForm center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="flex items-center pa3 f4 f4-ns lh-copy"
      >
        <button
          className="SubmitButton f3 dt"
          type="submit"
        >
          <div
            className="Symbol dtc"
          >
            +
          </div>
        </button>
        <input
          className="Input mw-100"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
</div>
`;
